### PR TITLE
Build the partitionConsumer into the topicConsumer

### PR DESCRIPTION
**Pull request by eapache**
@Shopify/kafka this is what I meant by merging them. Adding a flag to the topicConsumer makes the partitionConsumer obsolete.

### CODE REVIEW

File: tools/kafka-console-topicconsumer/kafka-console-topicconsumer.go [hunk 1]
```diff
@@ -110,6 +112,24 @@ func main() {
...
+
+	tmp := strings.Split(*partitions, ",")
+	var pList []int32
+	for i := range tmp {
+		val, err := strconv.ParseInt(tmp[i], 10, 32)
+		if err != nil {
+			return nil, err
+		}
+		pList = append(pList, int32(val))
+	}
```
Comment (wvanbergen) [inline tools/kafka-console-topicconsumer/kafka-console-topicconsumer.go]: Should we check whether the partition is the `c.Partitions(*topic)` list, or just let it fail later?

**Comment by wvanbergen** [comment]
Maybe just call it `kafka-console-consumer` now?

File: tools/kafka-console-topicconsumer/kafka-console-topicconsumer.go (hunk 1 above)
Comment (eapache) [inline tools/kafka-console-topicconsumer/kafka-console-topicconsumer.go]: It fails later pretty nicely with `ERROR: Failed to start consumer for partition -1: kafka server: Request was for a topic or partition that does not exist on this broker.`

Comment (wvanbergen) [inline tools/kafka-console-topicconsumer/kafka-console-topicconsumer.go]: :+1: that should be enough :)

**Comment by eapache** [comment]
Oh, does this whole thing break semantic versioning? Does this need to be a 2.0 change?

**Comment by wvanbergen** [comment]
I don't feel it's needed, but maybe we should state this in the `tools/README`?

**Comment by eapache** [comment]
Added a note to `tools/README` that they are not guaranteed to version like the main library,

**Comment by eapache** [comment]
The alternative would just be to not remove `partitionConsumer`, since `topicConsumer` hadn't been released yet

**Comment by wvanbergen** [comment]
Like I said, I don't feel too strongly about maintaining backwards compatibility for the tools, but if you do that works for me :+1:

**Comment by eapache** [comment]
I'm not 100% sure I want to commit to API stability for the tools for eternity, but in this case it's easy enough. I'll add "remove partitionConsumer" to our 2.0 wiki page.

**Comment by eapache** [comment]
Last :eyes:? I had to do some git dancing to rebase while undeleting the partitionConsumer.

File: tools/README.md [hunk 2]
```diff
@@ -4,7 +4,7 @@ This folder contains applications that are useful for exploration of your Kafka
 Some of these tools mirror tools that ship with Kafka, but these tools won't require installing the JVM to function.
 
 - [kafka-console-producer](./kafka-console-producer): a command line tool to produce a single message to your Kafka custer.
-- [kafka-console-partitionconsumer](./kafka-console-partitionconsumer): a command line tool to consume a single partition of a topic on your Kafka cluster.
-- [kafka-console-topicconsumer](./kafka-console-topicconsumer): a command line tool to consume all partition of a topic on your Kafka cluster.
+- [kafka-console-partitionconsumer](./kafka-console-partitionconsumer): (deprecated) a command line tool to consume a single partition of a topic on your Kafka cluster.
+- [kafka-console-consumer](./kafka-console-consumer): a command line tool to consume arbitrary partitions of a topic on your Kafka cluster.
```
Comment (wvanbergen) [inline tools/README.md]: "all or an arbitrary list of"?

**Comment by wvanbergen** [comment]
minor wording quibble - :shipit:

File: tools/README.md (hunk 2 above)
Comment (eapache) [inline tools/README.md]: "all of them" still qualifies as an arbitrary list :)

**Pull request merged**